userspace/libsinsp/cri.cpp

```diff
@@ -400,22 +401,73 @@ bool cri_interface::is_pod_sandbox(const std::string &container_id)
 	return status.ok();
 }
 
-uint32_t cri_interface::get_pod_sandbox_ip(const std::string &pod_sandbox_id)
+void cri_interface::get_pod_info_cniresult(runtime::v1alpha2::PodSandboxStatusResponse &resp, std::string &cniresult)
+{
+	Json::Value root;
+	Json::Reader reader;
+	const auto &info_it = resp.info();
+	if(reader.parse(info_it.find("info")->second, root))
+	{
+		Json::Value jvalue;
+		/* Lookup approach is brute force "try all schemas" we know of, do not condition by container runtime for possible future "would just work" luck in case other runtimes standardize on one of the current schemas. */
```

Comment:
A standardization would be really appreciated here. Based on the example you mentioned in the PR, there was (for cri-o) a `cniVersion\":\"1.0.0` specified where I would assume that it follows the CNI specification. Maybe worth a follow-up.

Reply:
Great idea, created a ticket and left a comment.

Reply:
@incertum Thanks a lot!